javascript/webapp/src/utils/specificationCurveManagement.ts

```diff
@@ -38,6 +39,48 @@ export function row2spec(d: any): Specification {
 		c95Upper: d.upper_bound,
 		c95Lower: d.lower_bound,
 		refutationResult: d.refutation_result,
-		taskId: d.task_id,
 	}
 }
+
+export function buildOutcomeGroups(
+	specifications: Specification[],
+): Specification[] {
+	const grouped = groupBySpecification(specifications)
+
+	return grouped.sort(function (a, b) {
+		return a?.estimatedEffect - b?.estimatedEffect
+	})
+}
+function returnKeys(item: Specification) {
+	return [item.treatment, item.causalModel, item.estimator]
+}
+
+function returnGroupLetter(number: number) {
+	return String.fromCharCode(97 + number).toUpperCase()
+}
+
+function insertGroupsToObjects(groups: any) {
+	return Object.keys(groups).map((outcomeGroup: any, groupNumber: number) => {
+		return groups[outcomeGroup]
+			.sort((a: Specification, b: Specification) => {
+				//order by primary first?
+				return a?.outcome.localeCompare(b?.outcome)
+			})
+			.map((specification: any, specificationNumber: number) => {
+				return {
+					...specification,
+					id: returnGroupLetter(specificationNumber) + (groupNumber + 1), //So it doesn't start in 0
+				}
+			})
+	})
+}
+
+function groupBySpecification(array: Specification[]): Specification[] {
```

Comment:
I couldn't think of a better way of doing this.
Every specification is has 3 main shared things: treatment, causal model and estimator. if we have multiple outcomes, throughout them, the combination of these three things says that they're the same specification.
What I need is:

- For every different outcome, there's a letter.
- Every specification has a number.
I form the prop id with that
For the same specification in different outcomes, they must have the same Number. So A1 and B1 are the same specification, in different outcomes.
What I'm doing now is: Creating an object with all the existing keys (treatment, causal model and estimator) and having their specifications inside of them as an array.
They I go through these keys, and inside I find their different outcomes.
Each key becomes a number, and each outcome becomes a letter.